### ISSUE TITLE
Meta: Implement all qlog probes

### ISSUE THREAD

**Issue by agrover** [mozilla/neqo#528]
Once basic qlog support lands in Neqo (#526), we still need to add probes for all the things qlog can log. These should be *relatively* straightforward, and can likely be done as a group effort.

The below list is all event types supported by the `qlog` crate, version 0.2. Future versions of the qlog specification and the qlog crate may change this list. `qlog` supports "min" versions of many event constructors, which contain only the most commonly relevant fields of the event. Either version may be used.

- [ ] congestion_state_updated(_min)
- [ ] connection_id_updated(_min)
- [x] connection_started(_min)
- [ ] connection_state_updated(_min)
- [ ] datagram_dropped(_min)
- [ ] datagrams_received(_min)
- [ ] datagrams_sent(_min)
- [ ] frames_processed
- [ ] h3_data_moved(_min)
- [ ] h3_frame_created(_min)
- [ ] h3_frame_parsed(_min)
- [ ] h3_parameters_set(_min)
- [ ] h3_push_resolved(_min)
- [ ] h3_stream_type_set(_min)
- [ ] loss_timer_set(_min)
- [ ] marked_for_retransmit
- [ ] metrics_updated(_min)
- [ ] packet_buffered
- [ ] packet_dropped(_min)
- [ ] packet_lost(_min)
- [x] packet_received(_min)
- [x] packet_sent(_min)
- [ ] qpack_dynamic_table_updated
- [ ] qpack_headers_decoded(_min)
- [ ] qpack_headers_encoded(_min)
- [x] qpack_instruction_received(_min) (partial)
- [ ] qpack_instruction_sent(_min)
- [ ] qpack_state_updated(_min)
- [ ] qpack_stream_state_updated
- [ ] recovery_parameters_set(_min)
- [ ] server_listening(_min)
- [ ] ~~spinbit_updated~~ (Neqo does not implement spin bit)
- [ ] stream_state_updated(_min)
- [x] transport_parameters_set(_min)

**Comment by agrover** [comment]
Nice to have but implementing all of this is not critical.

**Comment by mansf-osk** [comment]
Hey, I could help here! Are there any [events in the currently used qlog version](https://docs.rs/qlog/0.13.0/qlog/events/index.html#modules) that would be especially useful to implement?

**Comment by larseggert** [comment]
Thanks for helping out! I think we first need to check if the list in the first comment is still accurate.

**Comment by mansf-osk** [comment]
Makes sense, I've compiled a new list below

## Up to date list of qlog events

*for qlog crate version [0.13.0](https://docs.rs/qlog/0.13.0/qlog/events/index.html) / draft versions 09 ([quic](https://datatracker.ietf.org/doc/html/draft-ietf-quic-qlog-quic-events.html#name-quic-event-overview), [h3](https://datatracker.ietf.org/doc/html/draft-ietf-quic-qlog-h3-events.html#name-introduction))*

### Connectivity events

- [ ] server_listening
- [x] connection_started
- [ ] connection_closed
- [ ] connection_id_updated
- [ ] spin_bit_updated
- [x] connection_state_updated
- [ ] path_assigned (not implemented in qlog crate)
- [ ] mtu_updated

### Transport events

- [x] version_information
- [ ] alpn_information
- [x] parameters_set
- [ ] parameters_restored
- [x] packet_sent
- [x] packet_received
- [x] packet_dropped
- [ ] packet_buffered
- [ ] packets_acked
- [ ] udp_datagrams_sent
- [ ] udp_datagrams_received
- [ ] udp_datagram_dropped
- [ ] stream_state_updated
- [ ] frames_processed
- [ ] stream_data_moved
- [ ] datagram_data_moved
- [ ] migration_state_updated

### Security Events

- [ ] key_updated
- [ ] key_discarded

### Recovery events

- [ ] recovery_parameters_set
- [x] recovery_metrics_updated
- [ ] congestion_state_updated
- [ ] loss_timer_updated
- [x] packet_lost
- [ ] marked_for_retransmit
- [ ] ecn_state_updated

### HTTP3 events

- [ ] parameters_set
- [ ] parameters_restored
- [ ] stream_type_set
- [ ] priority_updated (not implemented in qlog crate)
- [ ] frame_created
- [ ] frame_parsed
- [ ] datagram_created (not implemented in qlog crate)
- [ ] datagram_parsed (not implemented in qlog crate)
- [ ] push_resolved

### QPACK

*those have been [removed](https://github.com/quicwg/qlog/pull/335) from the rfc but are still implemented in the qlog crate*

- [ ] qpack_dynamic_table_updated
- [ ] qpack_headers_decoded
- [ ] qpack_headers_encoded
- [x] qpack_instruction_received (partial)
- [ ] qpack_instruction_sent
- [ ] qpack_state_updated
- [ ] qpack_stream_state_updated

**Comment by larseggert** [comment]
@mansf-osk I'd prefer editing the list in the first comment instead. At least for me it's easier to track things that way. @mxinden?
